keys/keys_test.go

```diff
@@ -188,6 +198,11 @@ func TestMetaReverseScanBounds(t *testing.T) {
 			expStart: Meta1Prefix,
 			expEnd:   Meta2Prefix.Next(),
 		},
+		{
+			key:      Meta2KeyMax,
+			expStart: Meta2Prefix,
+			expEnd:   Meta2KeyMax.Next(),
+		},
```

Comment:
we should have some error cases here and in the forward scan tests as well. Could you add `expError string` to the struct and put in tests that fail specific to the scan direction (test the error with `IsError`).